src/main/java/org/fcrepo/spec/testsuite/authz/WebACDefaultACLs.java

```diff
@@ -0,0 +1,134 @@
+        //create a child  and child acl with read privileges and no acl:default
+        final Response childPost = doPost(grandParentUri, headers, simpleRDFBody);
+        final String childUri = getLocation(childPost);
+        createAclForResource(childUri, "user-read-only-no-default.ttl", this.username);
+
+        //create a grandchild  with no acl.
+        final Response grandChildPost = doPost(childUri, headers, simpleRDFBody);
+        final String grandChildUri = getLocation(grandChildPost);
+
+        // verify that the user can read and write to the grandparent
+        doGet(grandParentUri, false);
+        doPost(grandParentUri, headers, simpleRDFBody, false);
+
+        // verify that the user can read but not write to the child
+        doPostUnverified(childUri, headers, simpleRDFBody, false).then().statusCode(clientErrorRange());
+        doGet(childUri, false);
+
+        //verify that the grandchild can read and write.
+        doGet(grandChildUri, false);
+        doPost(grandChildUri, false);
+    }
+
+    /**
+     * 5.9-B - In the case that the controlled resource is uncontained and has no ACL, or that there is no ACL at any
+     * point in the containment hierarchy of the controlled resource, then the server must supply a default ACL.
+     *
+     * @param uri of base container of Fedora server
+     */
+    @Test(groups = {"SHOULD"})
+    @Parameters({"param0"})
+    public void serverSuppliedDefaultAcl(final String uri) {
+        setupTest("5.9-C", "In the case that the controlled resource is uncontained and has no ACL, or " +
+                           "that there is no ACL at any point in the containment hierarchy of the " +
+                           "controlled resource, then the server must supply a default ACL.",
+                  "https://fedora.info/2018/06/25/spec/#inheritance", ps);
+
+        //retrieve the default acl resource link from the root resource
+        final String aclUri = getAclLocation(uri);
+        //GET the default acl and verify success
+        doGet(aclUri);
+    }
+
+    /**
+     * 5.9-D - Default ACL should be on the same server as the controlled resource.
+     *
+     * @param uri of base container of Fedora server
+     */
+    @Test(groups = {"SHOULD"})
+    @Parameters({"param0"})
+    public void defaultAclOnSameServer(final String uri) {
+        setupTest("5.9-D", "The default ACL resource should be located in the same server as the " +
+                           "controlled resource.",
+                  "https://fedora.info/2018/06/25/spec/#inheritance", ps);
+        //GET the default acl and verify that it is on the same server as the root uri.
+        final String aclUri = getAclLocation(uri);
+        assertEquals("The default ACL resource is not located in the same server as the controlled resource.",
+                     URI.create(uri).getHost(), URI.create(aclUri).getHost());
```

Comment:
Is host comparison sufficient? Should this condition pass if the ACL is at a different port?

Reply:
I suppose if by same server we mean same serving process, rather than host, I suppose we should check the host as well.